plugins/inputs/opcua_listener/opcua_listener.go

```diff
@@ -31,11 +31,14 @@ func (o *OpcUaListener) Init() (err error) {
 	return err
 }
 
-func (o *OpcUaListener) Gather(_ telegraf.Accumulator) error {
-	return nil
+func (o *OpcUaListener) Gather(acc telegraf.Accumulator) error {
+	if o.client.successfulConnect || o.SubscribeClientConfig.ConnectFailBehavior == "Ignore" {
+		return nil
+	}
+	return o.connect(acc)
 }
```

Comment:
This doesn't match what you state in the config. Here you use uppercase `Ignore` while in the config you use lowercase... I'm on the lowercase side, so please change the check. Furthermore, you should check the input value for `connect_fail_behavior` in `Init()` to make sure the user uses valid values!